res/controllers/Hercules-DJControl-Inpulse-500-script.js

```diff
@@ -0,0 +1,1566 @@
+    }
+  }
+}
+
+DJCi500.fxEnabledIndicator = function(_value, group, _control, _status) {
+  var deckA = DJCi500.deckA.currentDeck;
+  var deckB = DJCi500.deckB.currentDeck;
+  var active = engine.getValue(group, 'enabled');
+
+  if (group == "[QuickEffectRack1_" + deckA + "]") {
+    midi.sendShortMsg(0x96, 0x66, active ? 0x1C : 0x60)
+  } else if (group == "[QuickEffectRack1_" + deckB + "]") {
+    midi.sendShortMsg(0x97, 0x66, active ? 0x1C : 0x60)
+  }
+}
+
+DJCi500.Deck = function (deckNumbers, midiChannel) {
+  components.Deck.call(this, deckNumbers);
+  // Allow components to access deck variables
+  var deckData = this;
+
+  // For loop and looprolls
+  var fractions = ['0.125', '0.25', '0.5', '1', '2', '4', '8', '16'];
+  var shiftFractions = ['0.03125', '0.0625', '32', '64', '128', '256', '512', '512'];
+
+  // For beatjumps
+  var jumpValues = ['1', '1', '2', '2', '4', '4', '8', '8'];
+  var jumpValuesShift = ['16', '16', '32', '32', '64', '64', '128', '128'];
+
+  // Brake status for this deck
+  this.slowPauseSetState = [false, false, false, false];
+
+  // Vinyl button state
+  this.vinylButtonState = [DJCi500.initialVinylMode, DJCi500.initialVinylMode, DJCi500.initialVinylMode, DJCi500.initialVinylMode];
+
+  // Pitch ranges and status
+  this.pitchRanges = [0.08, 0.10, 0.15, 0.16, 0.24, 0.50, 0.90]; //select pitch range
+  this.pitchRangeId = 0; //id of the array, one for each deck
+
+  // Effect section components
+  this.effectEnabled = false;
+
+  // Make sure the shift button remaps the shift actions
+  this.shiftButton = new components.Button({
+    midi: [0x90 + midiChannel, 0x04],
+    input: function(_channel, _control, value, _status, _group) {
+      if (value === 0x7F) {
+        deckData.forEachComponent(function(component) {
+          if (component.unshift) {
+            component.shift();
+          }
+        });
+      } else {
+        deckData.forEachComponent(function(component) {
+          if (component.unshift) {
+            component.unshift();
+          }
+        });
+      }
+    },
+  });
+
+  this.loadButton = new components.Button({
+    midi: [0x90 + midiChannel, 0x0D],
+    shiftOffset: 3,
+    shiftControl: false,
+    shiftChannel: true,
+    sendShifted: true,
+    unshift: function () {
+      this.inKey = 'LoadSelectedTrack';
+    },
+    shift: function () {
+      this.inKey = 'eject';
+    },
+  });
+
+  // Transport section
+  // Play button, for some reason the group is not correct on this one?
+  this.playButton = new components.PlayButton({
+    midi: [0x90 + midiChannel, 0x07],
+    shiftOffset: 3,
+    shiftControl: false,
+    shiftChannel: true,
+    sendShifted: true,
+    unshift: function () {
+      this.input = function (channel, control, value, status, group) {
+        if (value === 0x7F) {
+          if (engine.getValue(deckData.currentDeck, "play_latched")) {      //play_indicator play_latched
+            var deck = parseInt(deckData.currentDeck.charAt(8));

```

Comment:
Suggestion: Replace custom code with built-in [`script.deckFromGroup` helper function](https://github.com/mixxxdj/mixxx/wiki/MIDI-Scripting#helper-functions).
```suggestion
            var deck = script.deckFromGroup(deckData.currentDeck);
```